pilot/test/integration/cloudfoundry/copilot_test.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration_test
```

Comment:
cc @nmittler 
I was thinking that the code in pilot/test can be structured as pilot/test/integration/kube/..., pilot/test/integration/cloudfoundry/... I don't see the code moves for the former.. Also, shouldn't this package be called cloudfoundry ?

Reply:
@rshriram we can discuss separately after both PRs land

Reply:
IMHO, tests are independent from the platform already. What needs to be abstracted are:
1. deployment - we should consider leveraging helm or whatever is the convergence script to deploy istio
2. client invocation -- all usages of `kubectl run` need to be shielded from the test framework
3. log checking -- similarly, it's just one function, `kubectl logs` that needs to be pulled out.

For what it's worth, I managed to run a bunch of tests locally, with a bunch of envoys as processes, pilot as process, etc, without any platform.